### ISSUE TITLE
`.from` is broken since v1.10 and earlier... much, much earlier

### ISSUE THREAD

**Issue by Cpt0r** [greensock/GreenSock-JS#244]
## TL;DR:
`.from` doesn't tween using up-to-date object values at point of execution but is using cached values at point of declaration.

**Tested versions:** 
`1.20.3` - latest
`1.10.0` - because why not
`1.542 beta` (what?!  20 results on google! 🤣  ...yea, we're planning to upgrade soon... I hope).

## Reproduction
Each section has a correspondingly numbered block here:
https://codepen.io/Cpt0r/pen/mqOypR

Uncomment the line in each block as you follow the steps to observe the effects.

### 1. [ works ] Tween from current position
<- Moves left.

### 2. [ works ] Offset then tween - uses updated position
Moves right ->.

### 3. [ breaks ] Introduce delay before offset - uses original position
<- Moves left again (unless `time == 0`).

### 4. [ fixes ] Offset before the delay - uses updated position again
Moves right again ->, however to `x: 500` not `x: 900` ie. the `.set` just before `.from` is still effectively ignored.

**4a:** `.set`ting within the same tween
**4b:** `.set`ting in a separate prior tween

**Note:** neither **4a** nor **4b** work in `v1.542 beta`.

### 5. Verify that `.to` and `.fromTo` work in all situations
While `.fromTo` in **5a** has no choice but to use explicitly provided starting and ending values, `.to` in **5b** must tween using the most up-to-date starting value of the object - which it does correctly, unlike `.from` above.

### 6. It gets weirder
This time comment out all the blocks going back to step **1**, our baseline.

What do you expect to happen when you uncomment **6**?  Here's the simple code:
```
tl()
  .to moo, 0.5, x: 200
  .from moo, 0.5, x: 100
```

Well, basically exactly what it says: 0 -> 200, then 100 -> 200, right?

Nope, you instead get: 100 -> 200, then 100.  But actually there's some artifact in there so it's more like: 100 -> 200, then 98 -> 100.

And in `v1.542 beta` its a jittery 0 -> 200 -> 100.

Of course uncommenting **2** does absolutely nothing, it is still ignored.

#### Even moar weirderer
Uncomment **3** and now you get: 100 -> 200, then 100 -> 0 - all we did was add an unrelated tween/delay!

#### Now for the really fun bit
Uncomment **4a** and behold: 100 -> 200, then 100 -> 500.

But we expected: 500 -> 200, then 100 -> 900.  Here's the code again for reference:

```
tl()
  .set moo, x: 500
  .to {}, 0.5, {}
  .to moo, 0.5, x: 200
  .set moo, x: 900
  .from moo, 0.5, x: 100
```

If we omit the `.from` call then everything behaves exactly as expected - you can even add a `.to` at the end or whatever.

**Comment by jackdoyle** [comment]
I had a tough time following all the comments/instructions, but I didn't see any unexpected behavior. I suspect it's just a misunderstanding about a few things...

First, please look at this: https://greensock.com/immediateRender

Also, keep in mind that the set() method on TimelineLite/Max will automatically set immediateRender:true only if it's at the VERY beginning of the timeline because that seems like the most intuitive behavior. It has immediateRender:false if it's at a non-zero time. That's why number 3 "broke" :)

And again, most people want from() tweens to render their starting values IMMEDIATELY (hence immediateRender:true is the default). I suspect that once you understand those two things, you'll be able to get exactly the behavior you want. You can set immediateRender:false on your from() tweens if you'd prefer the non-default behavior. 

Does that clear things up?

**Comment by Cpt0r** [comment]
Yep, didn't see that flag.  The irony of trying hard to make something clear and unobtuse is it often ends up being just that.  But it was really weird behaviour, so I had to make sure it was understood.

I think that flag should be mentioned at the very top of the documentation, not the very bottom, with a clear example of how it differs from `.set` and the like.  I'd never expect the above examples to do any of the whacky things they end up doing, nor conclude that its due to a simple flag - but it did smell like I'm missing something simple.

Thanks for the fast response.  And mentioning the `.set` semantics.
